docs/decisions/broker-trigger.md

```diff
@@ -0,0 +1,137 @@
+the exact mechanics of routing to specific consumers. Event producers and
+consumers are decoupled temporally but also through a deliberately black-box
+abstraction which can optimize the underlying routing layer dynamically.
+
+## Objects
+
+The following objects are the minimum set needed to experiment and collect
+customer feedback on the Eventing object model, aka an MVP (including rough YAML
+definitions; final naming and field definitions subject to change based on
+implementation experience).
+
+### Broker
+
+A Broker represents a "bucket of events". It is a namespaced object, with
+automation provided to provision a Broker named `default` in
+appropriately-labelled namespaces. It is expected that most users will not need
+to explicitly create a Broker and simply enabling eventing in the namespace to
+create the `default` Broker will be sufficient. Assuming that the Broker model
+is successful, future recommendations for Source CRD authors would be to default
+the Source's `spec.sink` attribute to point to the `default` Broker in the
+namespace.
+
+Historical storage and replay of events is not part of the Broker design or MVP.
+
+Broker implements the "Addressable" duck type to allow Source custom resources
+to target the Broker. Broker DNS names should also be predictable (e.g.
+`default-broker` for the `default` Broker), to enable whitebox event producers
+to target the Broker without needing to reconcile against the "Addressable" duck
+type.
+
+#### Broker Object definition
+
+```golang
+type Broker struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec BrokerSpec `json:"spec,omitempty"`
+	Status BrokerStatus `json:"status,omitempty"`
+}
+
+type BrokerSpec struct {
+  // Empty for now
+}
+
+type BrokerStatus struct {
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	Address duckv1alpha1.Addressable `json:"address,omitempty"`
+}
+```
+
+### Trigger
+
+A Trigger represents a registration of interest (request for delivery to an
+Addressable object or URL) in a filtered selection of events delivered to a
+Broker. For the MVP, Triggers may only target Addressable objects in the same
+namespace.
+
+#### Buffering
+
+Triggers are assumed to buffer (durably queue) filtered events from the Broker
+from the time they are created, and to deliver the buffered events to the target
+Addressable as it is able to receive them.
+
+#### Filtering
+
+An event consumer which is interested in several sets of events may need to
+create multiple Triggers, each with an ANDed set of filter conditions which
+select the events. In the case of multiple Triggers with the same target whose
+filter conditions select an overlapping set of events, events will be delivered
+once for each Trigger which selects the event (i.e. Trigger selection criteria
+are independent, not aggregated in the Broker).
```

Comment:
How is this uniqueness determined? Some discussions in CloudEvents about this one.

Reply:
I don't understand the question. What 'uniqueness' is being determined?

I think what this section is trying to say is that conceptually, each event that enters the Broker will be tested against every Trigger independently. Thus if multiple Triggers match that incoming event, then that incoming event is sent to the Subscriber of all those Triggers. If some of those Triggers have the same Subscriber, then that Subscriber will be sent the incoming event multiple times (once for every Trigger that matches it and has that Subscriber).

Reply:
Yes, what Adam said. I'll try to clear this up.